hs_core/decorators.py

```diff
@@ -0,0 +1,35 @@
+from functools import wraps
+from django.shortcuts import redirect
+from django.utils.decorators import method_decorator
+
+from theme.models import UserProfile
+
+from pprint import pprint
+
+def profile_required(view_func):
+    '''Checks all required UserProfile fields.'''
+    def _wrapped_view(request, *args, **kwargs):
+        pprint(request.__dict__)
+        profile = UserProfile.objects.filter(user=request.user).first()
+        profile_valid = (profile.middle_name and
+                            profile.state and
+                            profile.zipcode and
+                            profile.country and
+                            profile.date_of_birth and
+                            profile.last_four_ss and
+                            profile.phone_1)
```

Comment:
I *think* this `profile_valid` variable could cause problems b/c your `profile` variable might be `None` (if `request.user` is not set).

Reply:
ah...and I guess in conjunction this will never happen b/c the `login_required` will fail first...